src/rules/tryceratops/rules/reraise_no_cause.rs

```diff
@@ -0,0 +1,32 @@
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{Stmt, StmtKind};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct ReraiseNoCause;
+);
+impl Violation for ReraiseNoCause {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use `raise from` to specify exception cause")
+    }
+}
+
+/// TRY200
+pub fn reraise_no_cause(checker: &mut Checker, body: &[Stmt]) {
+    for stmt in body.iter() {
+        if let StmtKind::Raise { cause, .. } = &stmt.node {
+            if cause.is_none() {
+                checker.diagnostics.push(Diagnostic::new(
+                    ReraiseNoCause,
+                    Range::from_located(stmt),
+                ));
```

Comment:
I think we need to recurse here to handle cases like:

```py
def bad():
    try:
        process()
    except MyException:
        if True:
            raise MainFunctionFailed()
```

This came up in another PR -- here's my explanation: https://github.com/charliermarsh/ruff/pull/2073#discussion_r1083345058

Reply:
(Otherwise, looks good!)